src/cmd/extrinsics/call.rs

```diff
@@ -34,17 +36,25 @@ use jsonrpsee::{
     rpc_params,
     ws_client::WsClientBuilder,
 };
+use pallet_contracts_primitives::{
+    ContractResult,
+    ExecReturnValue,
+};
 use serde::Serialize;
 use sp_core::Bytes;
-use std::fmt::Debug;
+use std::{
+    fmt::Debug,
+    result,
+};
 use subxt::{
     rpc::NumberOrHex,
     ClientBuilder,
     Config,
     DefaultConfig,
 };
 
-type ContractExecResult = pallet_contracts_primitives::ContractExecResult<Balance>;
+type ContractExecResult =
+    ContractResult<result::Result<ExecReturnValue, RuntimeDispatchError>, Balance>;
```

Comment:
This is the crux of the change, it is using the `subxt` generated `RuntimeDispatchError` instead of the default `sp_runtime::DispatchError`. This means it is using the same `DispatchError` type as would be returned in case of an error on calling the extrinsic.